### PR TITLE
DOC: Add code examples to `jax.scipy.stats.mode` docs

### PR DESCRIPTION
**Pull request by rajasekharporeddy**

### CODE REVIEW

File: jax/_src/scipy/stats/_core.py [hunk 1]
```diff
@@ -46,6 +46,40 @@ def mode(a: ArrayLike, axis: int | None = 0, nan_policy: str = "propagate", keep
...
+    >>> mode, count = jax.scipy.stats.mode(x)
+    >>> mode, count
+    (Array(4, dtype=int32), Array(3, dtype=int32))
+
+    For multi dimensional arrays, ``jax.scipy.stats.mode`` compuptes the ``mode``
+    and the corresponding ``count`` along ``axis=0``:
+
+    >>> x1 = jnp.array([[1, 2, 1],
+    ...                 [3, 1, 3],
+    ...                 [1, 2, 2]])
```
Comment (jakevdp) [inline jax/_src/scipy/stats/_core.py:61]: The axis behavior would be clearer if you used a non-square array here.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/scipy/stats/_core.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/scipy/stats/_core.py]: Thanks! Updated the examples with non-square matrix.

**Comment by jakevdp** [comment]
Please squash the changes into a single commit - thanks!

**Comment by rajasekharporeddy** [comment]
Thanks! Squashed into single commit.

**Review by jakevdp — APPROVED** [review]
Thanks!
